### DISCUSSION TITLE
All categories using zia_url_categories data source

### DISCUSSION THREAD

**Discussion by aaron-spencer** [zscaler/terraform-provider-zia#156] [General]
I was wondering if there is a way to retrieve all URL Categories using the zia_url_categories data source, similar to the /urlCategories API endpoint. The data source description seems to suggest that all categories can be fetched, but I am only successful when providing a valid id value.

**Comment by willguibr** [comment]
Hi @aaron-spencer
Thanks for reaching out. At the moment, the provider only allows to retrieve individual categories. In future versions we plan to allow for the retrieval of all categories. Can you describe what would be your use case to retrieve all categories in single data source?

Thank you

**Reply by aaron-spencer** [reply]
Initially, it would be useful for building our Terraform configuration files, as our ZIA environments have been in production for some time and we are just starting to look at Terraform. As we move the configuration to be fully Terraform managed, it would likely serve more of a documentation/audit purpose, to detect changes outside of process.

**Reply by willguibr** [reply]
Hi @aaron-spencer understood. If the purpose is to move your ZIA configuration to Terraform compliant code, I'd recommend that you take a look at our [Zscaler-Terraformer](https://github.com/zscaler/zscaler-terraformer) tool. The tool is designed to import your entire ZIA and/or ZPA configuration by creating not only the state file but also the HCL itself.
You can watch the introduction video in our Zenith Community [Here](https://community.zscaler.com/t/video-zscaler-terraformer-migration-tool-launch/19684?u=wguilherme)
Give it a try and let me know if you have any questions. Either way I'll be looking at your request and see what we can do as it will require some SDK updates as well.
FYI: Feel free to reach out via email if needed: wguilherme@you_know_the_rest.com 😉

Thank you

William
